bokeh/models/mappers.py

```diff
@@ -63,3 +65,229 @@ class LinearColorMapper(ColorMapper):
+    same index.  This argument is optional.  If it is
+    not provided, the min and max of the data will be used.
+    """)
+
+    alpha = Seq(Float, help="""
+    A python array, the same length as the dictionary with the
+    desired alpha values
+    """
+    )
+
+    interpolationMethod = String(default='linear', help="""
+    A string which describes the method of interpolation.  The default, 'linear',
+    will result in a gradual transition from one color to the next.  Other
+    options are 'step' which will result in a single color for all values 
+    which fall between the segment markers (an effective floor function) 
+    """)
+
+    def __init__(self, palette=None, segments=None, alpha=1, interpolationMethod='linear', **kwargs):
+        # Elegantly handle the possability that some parameters can be passed from the
+        # kwargs structure (because of subclassing)
+        if palette is None and 'palette' in kwargs.keys():
+            palette = kwargs['palette']
+
+        if segments is None and 'segments' in kwargs.keys():
+            segments = kwargs['segments']
+
+        if alpha is None and 'alpha' in kwargs.keys():
+            alpha = kwargs['alpha']
+
+        if interpolationMethod is None and 'interpolationMethod' in kwargs.keys():
+            interpolationMethod = kwargs['interpolationMethod']
+
+
+        # Define a dict to hold any alpha values that we might extract
+        # from the different colors in the passed palette.  These can
+        # come from a 4-tuple or a
+        alpha_hold = {}
+
+        if palette is not None:
+            tmp_palette = expandPalette(palette)
+            palette = tmp_palette['palette']
+            alpha_hold = tmp_palette['alpha_hold']
+            kwargs['palette'] = palette
+
+        if alpha is not None and palette is not None:
+            if not isinstance(alpha, list):
+                # Convert the scalar object to a vector so that the Seq property
+                # type can be used.
+                alpha = [alpha] * len(palette)
+
+            # Overwrite the alpha values with those specified by the colors in the palette.
+            # This may not be the desired functionality.  Who should win, the information in
+            # the palette, or the direct alpha information specified in the alpha parameter?
+            for k,v in list(alpha_hold.items()):
+                alpha[k] = v
+
+            # Check to ensure that all of the alpha values are ok.
+            if any([x > 1 for x in alpha]):
+                raise ValueError('alpha can not have any values greater than 1.0')
+
+            if any([x < 0 for x in alpha]):
+                raise ValueError('alpha cannot have any values less than 0.0')
+
+            kwargs['alpha'] = alpha
+
+        if segments is not None: kwargs['segments'] = segments
+
+        kwargs['interpolationMethod'] = interpolationMethod
+
+        super(SegmentedColorMapper, self).__init__(**kwargs)
```

Comment:
I think this `__init__` will require some discussion. In general, almost all of our models have no initializers, they are intended to be declarative, with our properties type system handling validation and serialization.